.github/workflows/master.yml

```
@@ -44,7 +44,7 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version: 1.21
+          go-version-file: "go.mod"
```

As per [setup-go](https://github.com/marketplace/actions/setup-go-environment#getting-go-version-from-the-gomod-file)
```suggestion
          go-version-file: 'go.mod'
```